### PR TITLE
Syntax highlight code blocks in markdown.

### PR DESCRIPTION
**Pull request by kadams54**
Also set data-manual on script tags so that Prism doesn't scan the entire DOM for stuff to highlight.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4308/
Test PASSed.

**Comment by kadams54** [comment]
## QA

Here's the tricky thing: even though we now support syntax highlighting, you have to specify a language in the code block (which we didn't support before) in order to trigger the syntax highlighting. So... you can either QA this with a local charm and a README of your own making, or you can set a breakpoint and fake like a code block had the language set.

### Local Charm

You're on your own for that. Sorry.

### Faking Out the JS:

1. Set a breakpoint in app.js, at line 6534: `var language = Prism.languages[lang];`.
2. Load `/?store=trusty%2Fmongodb`.
3. When the breakpoint triggers, watch for when the `code` var is `"environment: amazon..."` - this should be the third time the breakpoint triggers.
4. Run this in the console: `lang = 'yaml'`.
5. Disable the breakpoint and continue - the page should now finish loading.
6. When you scroll down to the `juju status` output, it should now have YAML syntax highlighting.

File: jujugui/templates/index.html.mako [hunk 1]
```diff
@@ -336,12 +336,12 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
       'use strict' and d3 doesn't work under strict.
     -->
     % if raw:
-    <script src="${convoy_url}?app/assets/javascripts/react-with-addons.js&app/assets/javascripts/react-dom.js&app/assets/javascripts/classnames.js&app/assets/javascripts/clipboard.js&app/assets/javascripts/react-click-outside.js&app/assets/javascripts/ReactDnD.min.js&app/assets/javascripts/ReactDnDHTML5Backend.min.js&app/assets/javascripts/marked.js&app/assets/javascripts/prism.js&app/assets/javascripts/prism-languages.js"></script>
+    <script data-manual src="${convoy_url}?app/assets/javascripts/react-with-addons.js&app/assets/javascripts/react-dom.js&app/assets/javascripts/classnames.js&app/assets/javascripts/clipboard.js&app/assets/javascripts/react-click-outside.js&app/assets/javascripts/ReactDnD.min.js&app/assets/javascripts/ReactDnDHTML5Backend.min.js&app/assets/javascripts/marked.js&app/assets/javascripts/prism.js&app/assets/javascripts/prism-languages.js"></script>
```
Comment (hatched) [inline jujugui/templates/index.html.mako]: what is data-manual?

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -989,6 +989,15 @@ YUI.add('juju-gui', function(Y) {
       var state = this.state;
       var utils = views.utils;
       var charmstore = this.get('charmstore');
+      // Configure syntax highlighting for the markdown renderer.
+      marked.setOptions({
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: So we don't have to trigger it to render after content has been added to the dom?

Comment (kadams54) [inline jujugui/static/gui/src/app/app.js]: No. It's highlighted as part of the markdown render that is called in the entity-content-readme component.

File: jujugui/templates/index.html.mako (hunk 1 above)
Comment (kadams54) [inline jujugui/templates/index.html.mako]: data-manual tells Prism to not automatically scan the page for code blocks to highlight; that we'll trigger the highlighting manually. I mention as much in my commit message, but I'll put a comment here as well.

**Comment by anthonydillon** [comment]
Code :ok: QA :+1: Thanks @kadams54 looks great. Please do comment on the use of data-manual as its not clear on a combined script.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4315/
Test PASSed.

**Comment by hatched** [comment]
:+1:

**Comment by kadams54** [comment]
Thanks all! :shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4316/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**